spec/Handler/AgreementHandlerSpec.php

```diff
@@ -80,4 +81,86 @@ function it_handle_agreements_correctly(
 
         $this->handleAgreements($collection, 'checkout_form', $order, $shopUser);
     }
+
+    function it_should_throw_exception_when_no_agreements_returned(
+        Collection $collection,
+        OrderInterface $order,
+        ShopUserInterface $shopUser,
+        AgreementRepositoryInterface $agreementRepository
+    ): void {
+        $agreementRepository->findAgreementsByContext('checkout_form')->willReturn([]);
+
+        $this->shouldThrow(NotFoundHttpException::class)
```

Comment:
Why throwing HTTP exception inside bussines logic? I think we should throw dedicated exception for bussines logic and then catch it in controller and then throw not found

Reply:
We decided to not to do this as we have no custom controller for the form handling.